### PR TITLE
[SILInliner] Added lexical lifetimes for arguments.

### PR DESCRIPTION
**Pull request by nate-chandler**
When the -enable-experimental-lexical-lifetimes flag is passed, added lexical borrow scopes for every non-trivial argument when inlining a function.

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please test

File: lib/SILOptimizer/Utils/SILInliner.cpp [hunk 1]
```diff
@@ -424,17 +424,39 @@ SILInlineCloner::cloneInline(ArrayRef<SILValue> AppliedArgs) {
...
     SILValue callArg = p.value();
     unsigned idx = p.index();
-    // Insert begin/end borrow for guaranteed arguments.
-    if (idx >= calleeConv.getSILArgIndexOfFirstParam() &&
-        calleeConv.getParamInfoForSILArg(idx).isGuaranteed()) {
-      if (SILValue newValue = borrowFunctionArgument(callArg, Apply)) {
-        callArg = newValue;
+    if (idx >= calleeConv.getSILArgIndexOfFirstParam()) {
+      if (Apply.getFunction()->hasOwnership() &&
+          Apply.getFunction()
```
Comment (atrick) [inline lib/SILOptimizer/Utils/SILInliner.cpp:435]: To make the code readable, I would create a local variable at the top of this function called `EnableLexicalLifetimes`

**Review by atrick — APPROVED** [review]
Looks good

**Comment by nate-chandler** [comment]
@swift-ci please test

File: lib/SILOptimizer/Utils/SILInliner.cpp (hunk 1 above)
Comment (nate-chandler) [reply lib/SILOptimizer/Utils/SILInliner.cpp]: Fixed!

**Comment by nate-chandler** [comment]
@swift-ci please test macos platform

**Pull request merged**